napari/layers/base/base.py

```diff
@@ -999,6 +1003,111 @@ def _data_to_world(self) -> Affine:
+                dims_point: the indices for the data in view
+
+        Returns
+        -------
+        start_point : np.ndarray
+            The point on the axis-aligned data bounding box that the cursor click
+            intersects with. This is the point closest to the camera.
+            The point is the full nD coordinates of the layer data.
+            If the click does not intersect the axis-aligned data bounding box,
+            an emtpy numpy array is returned (i.e., np.empty([]).
+        end_point : np.ndarray
+            The point on the axis-aligned data bounding box that the cursor click
+            intersects with. This is the point farthest from the camera.
+            The point is the full nD coordinates of the layer data.
+            If the click does not intersect the axis-aligned data bounding box,
+            an emtpy numpy array is returned (i.e., np.empty([]).
+        """
+        # create a mask to select the in view dimensions
+        dims_displayed = event.dims_displayed
+        dims_displayed_mask = np.zeros_like(event.dims_point, dtype=bool)
+        dims_displayed_mask[dims_displayed] = True
+
+        # create the bounding box in data coordinates
+        bbox = self._display_bounding_box(dims_displayed_mask)
+
+        # get the view direction in data coords (only displayed dims)
+        view_dir_world = event.view_direction
+        view_dir = np.asarray(self.vector_world_to_data(view_dir_world))[
+            dims_displayed_mask
+        ]
+
+        # Get the clicked point in data coords (only displayed dims)
+        click_pos_data = np.asarray(self.world_to_data(event.position))[
+            dims_displayed_mask
+        ]
+
+        # Determine the front and back faces
+        front_face_normal, back_face_normal = find_front_back_face(
+            click_pos_data, bbox, view_dir
+        )
+
+        # Get the locations in the plane where the ray intersects
+        if front_face_normal is not None and back_face_normal is not None:
+            start_point_disp_dims = (
+                intersect_ray_with_axis_aligned_bounding_box_3d(
+                    click_pos_data, view_dir, bbox, front_face_normal
+                )
+            )
+            end_point_disp_dims = (
+                intersect_ray_with_axis_aligned_bounding_box_3d(
+                    click_pos_data, view_dir, bbox, back_face_normal
+                )
+            )
+
+            # add the coordinates for the axes not displayed
+            start_point = np.asarray(event.dims_point)
+            start_point[dims_displayed_mask] = start_point_disp_dims
+            end_point = np.asarray(event.dims_point)
+            end_point[dims_displayed_mask] = end_point_disp_dims
+
+        else:
+            # if the click doesn't intersect the data bounding box,
+            # return empty points
+            start_point = np.empty((0,))
+            end_point = np.empty((0,))
```

Comment:
Is this advantageous vs "None"? Or maybe `np.full(3, np.nan)`? ie I'm thinking either we should respect the type (including the dimensionality), or use the "standard" Python "Optional" value.

Not feeling super strongly about this, it's just 0D arrays are often annoying to handle. =)

Reply:
I initially had `None`, but then I went with the empty array becausethat's what our layer data are when the layer doesn't have data. I am fine either way as long as it is clearly documented!